contracts/protocol/Agreement/EscrowedAgreement.sol

```diff
@@ -6,35 +6,59 @@ import "../Measurements/MeasurementsOnChain.sol";
 import "../Requirements/Requirements.sol";
 import "../Validator/Validator.sol";
 import "../Market/Offer.sol";
+import "../Profile/Profile.sol";
+import "../Utils/Ownable.sol";
 
-contract EscrowedAgreement is Agreement {
+contract EscrowedAgreement is Agreement, Ownable {
 
     address public buyer;
     address public seller;
     uint public amount;
+    uint public quantity;
+    Offer public offer;
     ERC20Protocol token;
 
     modifier onlyBuyer() {
         require(msg.sender == buyer);
         _;
     }
 
-    function EscrowedAgreement(ERC20Protocol _token, Offer _offer, uint _quantity) {
-        buyer = msg.sender;
+    modifier onlyStage(Stages _state) {
+        require(stage == _stage);
+        _;
+    }
+
+    enum Stages {
+        New,
+        InProgress,
+        Complete,
+        Canceled,
+        Reimbursed
+    }
+
+    Stages public stage;
+
+    function EscrowedAgreement(ERC20Protocol _token, Offer _offer, uint _quantity, address _buyer) {
+        buyer = _buyer;
         token = _token;
         amount = _offer.pricePerPackage()*_quantity;
         seller = _offer.seller();
+        offer = _offer;
+        quantity = _quantity;
+        stage = Stages.New;
     }
 
-    function escrowWithSeller() onlyBuyer returns (bool){
-        assert(token.transferFrom(buyer, this, amount));
+    function escrowWithSeller() onlyOwner onlyStage(Stages.New){
```

Comment:
I guess function is unnecessary, we can just go to InPogress state from constructor?